sklearn/tree/tests/test_tree.py

```diff
@@ -1266,3 +1268,39 @@ def check_min_weight_leaf_split_level(name):
 def test_min_weight_leaf_split_level():
     for name in ALL_TREES:
         yield check_min_weight_leaf_split_level, name
+
+
+def check_public_apply(name):
+    X_small32 = X_small.astype(tree._tree.DTYPE)
+
+    est = ALL_TREES[name]()
+    est.fit(X_small32, y_small)
+    assert_array_equal(est.apply(X_small32),
+                       est.tree_.apply(X_small32))
+
+
+def check_public_apply_sparse(name):
+    X_small32 = csr_matrix(X_small.astype(tree._tree.DTYPE))
+
+    est = ALL_TREES[name]()
+    est.fit(X_small32, y_small)
+    assert_array_equal(est.apply(X_small32),
+                       est.tree_.apply(X_small32))
+
+
+def test_public_apply():
+    for name in ALL_TREES:
+        yield (check_public_apply, name)
+
+    for name in SPARSE_TREES:
+        yield (check_public_apply_sparse, name)
+
+
+def check_apply_error(name):
+    est = ALL_TREES[name]()
+    assert_raises(ValueError, est.apply, X_small)
```

Comment:
Could this test's name be made more explicit? Is it supposed to fail because `est` is not fitted yet?